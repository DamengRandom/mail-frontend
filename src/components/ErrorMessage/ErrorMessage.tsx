export type tError = {
  className?: string,
  message: string,
}

export default function ErrorMessage({ className, message }: tError) {
  return (
    <p className={className}>{message}</p>
  )
};
