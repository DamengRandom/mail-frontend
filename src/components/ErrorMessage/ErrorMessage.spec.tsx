// import renderer from 'react-test-renderer';
import { cleanup, render, screen } from '@testing-library/react'
import ErrorMessage, { tError } from "./ErrorMessage";

describe('<ErrorMessage /> test', () => {
  let componentProps: JSX.IntrinsicAttributes & tError,
      componentContainer: HTMLElement;

  beforeEach(() => {
    // Arrange
    const props: tError = {
      className: "message",
      message: 'test message',
    };
    componentProps = props;
    // Act
    const { container } = render(<ErrorMessage {...componentProps} />);
    componentContainer = container;
  });

  afterEach(cleanup);

  it('snapshot test for ErrorMessage component', () => {
    // Assert
    expect(componentContainer.firstChild).toMatchSnapshot();
  });

  it('should render props for ErrorMessage component', async () => {
    // Arrange
    const newErrorMessage = "new error test message";
    componentProps = {
      ...componentProps,
      message: newErrorMessage,
    };
    // Act
    const { container } = render(<ErrorMessage {...componentProps} />);
    const errorMessage = await screen.findAllByText(newErrorMessage);
    // Assert
    expect(errorMessage).toBeDefined();
    expect((container.firstChild as HTMLInputElement).textContent).toEqual(newErrorMessage);
  });
});
