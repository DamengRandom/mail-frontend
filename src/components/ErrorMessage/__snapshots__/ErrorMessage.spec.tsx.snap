// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ErrorMessage /> test snapshot test for ErrorMessage component 1`] = `
<p
  class="message"
>
  test message
</p>
`;
